sr/comp/comp.py

```diff
@@ -36,6 +37,46 @@ def load_scorer(root: Path) -> ScorerType:
     return cast(ScorerType, score['Scorer'])
 
 
+def load_ranker(root: Path) -> RankerType:
+    """
+    Load the ranker module from Compstate repo.
+
+    :param Path root: The path to the compstate repo.
+    """
+
+    # Deep path hacks
+    score_directory = root / 'scoring'
+    ranker_source = score_directory / 'ranker.py'
+
+    if not ranker_source.exists():
+        # By default we support using the `league-ranker` package without
+        # modifications.
+        return ranker.LeagueRanker
+
+    saved_path = copy(sys.path)
+    sys.path.insert(0, str(score_directory))
+
+    try:
+        score = runpy.run_path(str(ranker_source))
+    finally:
+        sys.path = saved_path
+
+    ranker_class = cast(RankerType, score['Ranker'])
```

Comment:
Could we use a function rather than a class?

Reply:
While at the moment there's only one member on the class, there's a consideration that in future we may add others. Not sure if this is YAGNI, though I originally built this with two functions and then decided against actually supporting the other one for now.

Reply:
Other considerations from chatting:
- the class here really is just behaving as a namespace, which the module is probably better as
- if we wanted to pass in additional config to the `__init__`, say from other aspects of the compstate, that would be a breaking change if we'd gone with a function here
  - counter: we could pass such in as an argument, but that might itself be breaking and might require some dancing (with e.g: [`inspect`](https://docs.python.org/3/library/inspect.html)) to make work